packages/vscode-extension/src/debug/prerequisitesHandler.ts

```diff
@@ -19,9 +39,27 @@ export async function checkAndInstall(): Promise<Result<any, FxError>> {
       // ignore telemetry error
     }
 
+    const failures: CheckFailure[] = [];
     const localEnvManager = new LocalEnvManager(VsCodeLogInstance, ExtTelemetry.reporter);
     // TODO: LocalEnvManager deps
-    // TODO: vsc-related deps, e.g., login
+
+    // login
+    const accountFailure = await checkM365Account();
+    if (accountFailure) {
+      failures.push(accountFailure);
+    }
+
+    // handle failures
+    if (failures.length > 0) {
+      await handleFailures(failures);
+      return err(
```

Comment:
`return` here will lose telemetry, `throw` instead?

Reply:
Good point. Updated to `return` and add checkers information to error message as well, thus we can know which ones are failed.